### ISSUE TITLE
Java Module support

### ISSUE THREAD

**Issue by cowwoc** [tensorflow/tensorflow#39378]
**Describe the problem**
Applications using Java9+ benefit from libraries have Java Module (JPMS) support. This comes in two stages:

1. Ideally, add `module-info.java` to the library. Either publish a separate artifact for Java9+ or simply add `META-INF/versions/9/module-info.class` to support both versions simultaneously (multi-release jar)
2. If the library has split packages, you cannot add `module-info.java` yet in which case you can simply add `Automatic-Module-Name` to the manifest file and begin resolving the split package issue.

* Multi-release JARs
  * https://openjdk.java.net/jeps/238
  * https://blog.codefx.org/tools/multi-release-jars-multiple-java-versions/

**Comment by gadagashwini** [comment]
@cowwoc,
Does this help https://www.tensorflow.org/install/lang_java_legacy#tensorflow_with_the_jdk.Thanks!

**Comment by cowwoc** [comment]
@gadagashwini What are you asking? Did you add module-info.java to Tensorflow?

**Comment by gadagashwini** [comment]
@cowwoc, As you can see https://www.tensorflow.org/install/lang_java_legacy#tensorflow_with_the_jdk Tensorflow has Java compiler. Thanks!

**Comment by cowwoc** [comment]
@gadagashwini Sure, but this issue is about "Java Module" support which is about a lot more than just providing Java support.

**Comment by mohantym** [comment]
Hi @cowwoc !
I could locate [module-info.java](https://github.com/tensorflow/java/blob/master/tensorflow-framework/src/main/java/module-info.java) here in [Tensorflow/Java](https://github.com/tensorflow/java) repo.

Could you check the run time from [Tensorflow/Java ](https://github.com/tensorflow/tensorflow/issues/56892#issuecomment-1295230384)for your usecase.

Thank you!

**Comment by cowwoc** [comment]
Sorry @mohantym I'm not working on any AI projects at the moment. Can anyone else test?

**Comment by cowwoc** [comment]
I don't think you understand what Java Module (JPMS) is. It's not enough to say that Tensorflow/Java can run under a JVM. You'll need to create a new application that uses JPMS and import TensorFlow in your `module-info.java` and check for errors at compile-time and runtime.

**Comment by mohantym** [comment]
@cowwoc !

Thanks for the update. will let you know once i have some update on above pointers.

Thank you!

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.

**Comment by cowwoc** [comment]
This issue is still relevant.

**Comment by Venkat6871** [comment]
Hi @cowwoc ,
Thank you so much for filing this feature request! We truly appreciate your support. At the moment, this request is not prioritized and mark it for community contributions. If you would like to contribute on a fix, please feel free to fork the repository and submit a pull request in accordance with the TensorFlow [CONTRIBUTION GUIDELINES](https://www.tensorflow.org/community/contribute/code) and link this issue in your PR to ensure the maintainers can easily review and collaborate.To avoid duplicate work, please leave a comment here if you decide to take it on!
Thank you again for your interest and we look forward to seeing your contributions!
